### PR TITLE
mgr/dashboard: open api swagger in new tab

### PR DESCRIPTION
**Pull request by Pegonzal**
Fixes: https://tracker.ceph.com/issues/62350

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by nizamial09 — APPROVED** [review]

**Comment by Pegonzal** [comment]
jenkins test make check

**Comment by Pegonzal** [comment]
jenkins test make check

**Comment by cloudbehl** [comment]
@Pegonzal can you remove the commit, I think we can merge it

**Comment by Pegonzal** [comment]
jenkins test api

**Comment by cloudbehl** [comment]
jenkins test make check

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/dashboard-help/dashboard-help.component.html [hunk 1]
```diff
@@ -15,10 +15,10 @@
...
        target="_blank"
        i18n>documentation</a>
-    <button ngbDropdownItem
-            routerLink="/api-docs"
-            target="_blank"
-            i18n>API</button>
+    <a ngbDropdownItem
+       routerLink="/api-docs"
+       target="_blank"
+       i18n>API</a>
```
Comment (cloudbehl) [inline src/pybind/mgr/dashboard/frontend/src/app/core/navigation/dashboard-help/dashboard-help.component.html:21]: I think we should be adding the external Icon as well.


```suggestion
    <a ngbDropdownItem
       routerLink="/api-docs"
       target="_blank"
       i18n>API
         <i class="fa fa-external-link"></i>
       </a>
```

Comment (Pegonzal) [reply src/pybind/mgr/dashboard/frontend/src/app/core/navigation/dashboard-help/dashboard-help.component.html]: Added it both for documentation and API links

**Review by afreen23 — APPROVED** [review]
This works fine as I tested..thanks!
API takes to new tab! 

just one comment not related to this PR but I could not resist so commented!

File: src/pybind/mgr/dashboard/frontend/src/app/core/navigation/dashboard-help/dashboard-help.component.html [hunk 2]
```diff
@@ -14,11 +14,15 @@
        [ngClass]="{'disabled': !docsUrl}"
        href="{{ docsUrl }}"
        target="_blank"
-       i18n>documentation</a>
-    <button ngbDropdownItem
-            routerLink="/api-docs"
-            target="_blank"
-            i18n>API</button>
+       i18n>documentation
```
[the remark this answers is not in the document]
Comment (Pegonzal) [reply src/pybind/mgr/dashboard/frontend/src/app/core/navigation/dashboard-help/dashboard-help.component.html]: No idea... I don't think it matters but, I've changed it

**Review by cloudbehl — APPROVED** [review]

**Comment by nizamial09** [comment]
jenkins test dashboard

**Pull request merged**